### PR TITLE
[nl] Some cleanup for the url prompt handling

### PR DESCRIPTION
**Pull request by chejennifer**
- clean up code so that autorun and non autorun shares the logic for inputting queries from the url into the search box

screen-recording-_72_

### CODE REVIEW

File: static/js/apps/nl_interface/app.tsx [hunk 1]
```diff
@@ -34,6 +34,9 @@ export function App(): JSX.Element {
   const [contextList, setContextList] = useState<any[]>([]);
   const urlPrompts = useRef(getUrlPrompts());
   const autoRun = useRef(getUrlToken("a"));
+  const inputIntervalTimer = useRef(null);
+  const nextPromptDelayTimer = useRef(null);
+  const searchDelayTimer = useRef(null);
```
Comment (beets) [inline static/js/apps/nl_interface/app.tsx:39]: some documentation for these?

File: static/js/apps/nl_interface/app.tsx [hunk 2]
```diff
@@ -34,6 +34,9 @@ export function App(): JSX.Element {
   const [contextList, setContextList] = useState<any[]>([]);
   const urlPrompts = useRef(getUrlPrompts());
   const autoRun = useRef(getUrlToken("a"));
```
Comment (beets) [inline static/js/apps/nl_interface/app.tsx:36]: curious why autorun needs to be a ref? the value shouldn't change once the page has loaded (or rather, we won't need to listen to changes..)

**Review by beets — APPROVED** [review]
thanks - this is much cleaner!

File: static/js/apps/nl_interface/app.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/apps/nl_interface/app.tsx]: done

File: static/js/apps/nl_interface/app.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/apps/nl_interface/app.tsx]: oh yeah, this doesn't need to be a ref anymore, removed the ref part!

**Pull request merged**